src/isomorphic/classic/class/ReactClass.js

```diff
@@ -728,6 +728,11 @@ var ReactClassMixin = {
    * type signature and the only use case for this, is to avoid that.
    */
   replaceState: function(newState, callback) {
+    if (this.updater.isFiberUpdater) {
```

Comment:
I split them out in Stack because they look like they are conceptually linked but they're actually not since the callback isn't fired after a particular state update but after all the state updates. It's an API designed without consideration for batching.

At some point I considered making the enqueueCallback API public instead of the argument to setState.

We only have to do an extra flush now because they're synchronous but as soon as we get rid of that it won't cause an extra flush. They'all always be batched.

We shouldn't even need to traverse the parents in schedule update since we've already scheduled this node's priority.